boa/src/builtins/json/mod.rs

```diff
@@ -149,15 +153,17 @@ pub fn stringify(_: &mut Value, args: &[Value], ctx: &mut Interpreter) -> Result
         for field in fields {
             if let Some(value) = object
                 .get_property(&ctx.to_string(&field)?)
-                .map(|prop| prop.value.as_ref().map(|v| v.to_json()))
+                // TODO: handle the error case
+                .map(|prop| prop.value.as_ref().map(|v| v.to_json(ctx).unwrap()))
```

Comment:
This is a bit trickier, since it's an `Option::map()`, but we can just return the `to_json()`, and after the `map()`, we call `transpose()` and then use `?`, to propagate the exception.